LunrCoreLmdb/DelegatedIndex.cs

```diff
@@ -0,0 +1,450 @@
+            }
+
+            queryFactory(query);
+
+            for (int i = 0; i < query.Clauses.Count; i++)
+            {
+                Clause clause = query.Clauses[i];
+                Lunr.ISet<string> clauseMatches = Set<string>.Empty;
+
+                // Unless the pipeline has been disabled for this term, which is
+                // the case for terms with wildcards, we need to pass the clause
+                // term through the search pipeline. A pipeline returns an array
+                // of processed terms. Pipeline functions may expand the passed
+                // term, which means we may end up performing multiple index lookups
+                // for a single query term.
+                await foreach (string term in (clause.UsePipeline
+                    ? Pipeline.RunString(
+                        clause.Term,
+                        new TokenMetadata
+                        {
+                            { "fields", clause.Fields }
+                        },
+                        cancellationToken)
+                    : new[] { clause.Term }.ToAsyncEnumerable(cancellationToken)).WithCancellation(cancellationToken))
+                {
+                    // Each term returned from the pipeline needs to use the same query
+                    // clause object, e.g. the same boost and or edit distance. The
+                    // simplest way to do this is to re-use the clause object but mutate
+                    // its term property.
+                    clause = clause.WithTerm(term);
+
+                    // From the term in the clause we create a token set which will then
+                    // be used to intersect the indexes token set to get a list of terms
+                    // to lookup in the inverted index.
+                    var termTokenSet = TokenSet.FromClause(clause);
+                    IEnumerable<string> expandedTerms = IntersectTokenSets(termTokenSet).ToEnumeration();
+
+                    // If a term marked as required does not exist in the tokenSet it is
+                    // impossible for the search to return any matches.We set all the field
+                    // scoped required matches set to empty and stop examining any further
+                    // clauses.
+                    if (!expandedTerms.Any() && clause.Presence == QueryPresence.Required)
+                    {
+                        foreach (string field in clause.Fields)
+                        {
+                            requiredMatches.Add(field, Set<string>.Empty);
+                        }
+
+                        break;
+                    }
+
+                    foreach (string expandedTerm in expandedTerms)
+                    {
+                        // For each term get the posting and termIndex, this is required for building the query vector.
+                        InvertedIndexEntry posting = GetInvertedIndexEntryByKey(expandedTerm);
```

Comment:
You've got a bunch of possible null value warnings here. I know there are some in the original code as well, that I'll take care of, but it would be great to fix those as well before I merge